File: arduino_bot_ws/src/arduino_bot_test/scripts/simple_action_client.py

#!/usr/bin/env python3

import rospy
import actionlib
from arduino_bot_test.msg import FibonacciAction, FibonacciGoal


def fibonacci_client():
    client = actionlib.SimpleActionClient('fibonacci', FibonacciAction)
    client.wait_for_server()
    goal = FibonacciGoal(goal=20)
    print('sending goal order %s' % goal.goal)
    client.send_goal(goal)
    client.wait_for_result()
    return client.get_result()

if __name__ == '__main__':
    rospy.init_node('fibonacci_client')
    rospy.loginfo('simple action client started')
    result = fibonacci_client()
    print('result ', result.sequence)

File: arduino_bot_ws/src/arduino_bot_test/src/simple_action_client.cpp

#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include "arduino_bot_test/FibonacciAction.h"

int main(int argc, char** argv) 
{
    ros::init(argc, argv, "fibonacci_client");
    actionlib::SimpleActionClient<arduino_bot_test::FibonacciAction> action_client("fibonacci");

    ROS_INFO("Waiting for action server to start");
    action_client.waitForServer();

    ROS_INFO("Action server started, sending a goal");
    arduino_bot_test::FibonacciGoal goal;
    goal.goal = 20;

    action_client.sendGoal(goal);

    action_client.waitForResult(ros::Duration(30.0));
    
    auto result = action_client.getResult();
    ROS_INFO("Action finished:");
    
    for(int i : result->sequence) {
        ROS_INFO("%d", i);
    }

    return 0;
}


